api/utils/prompt/stdin.go

```diff
@@ -69,3 +74,36 @@ func NotifyExit() {
 	}
 	stdinMU.Unlock()
 }
+
+// EnableStdinTerminalFallback attempts to use a fallback like /dev/tty when stdin is
+// not the terminal. This can be useful when tsh is called by another terminal
+// tool (e.g. git).
+func EnableStdinTerminalFallback() {
+	stdinMU.Lock()
+	defer stdinMU.Unlock()
+	stdinTerminalFallback = true
+}
+
+func maybeUseStdinTerminalFallbackLocked(original *ContextReader) *ContextReader {
+	if !stdinTerminalFallback || runtime.GOOS == "windows" || original.IsTerminal() {
```

Comment:
Can you make `stdinTerminalFallback` an `sync/atomic.Bool` instead of protecting writes with a mutex? Should be cleaner and prevent races when reading and writing at the same time